tools/sigh.js

```diff
@@ -89,45 +113,37 @@ function railroad() {
   const {transform} = require('grammkit/lib/util');
   const handlebars = require('handlebars');
 
-  let renderTemplate = function(data, templatePath) {
-    let raw_template = fs.readFileSync(templatePath);
-    let template = handlebars.compile(raw_template.toString());
-    return template(data);
-  };
+  let diagramStyle = 'node_modules/grammkit/app/diagram.css';
+  let appStyle = 'node_modules/grammkit/app/app.css';
+  let baseTemplate = 'node_modules/grammkit/template/viewer.html';
 
-  for (let [grammarFile, _a, railroadFile] of sources.peg) {
-    let grammar = fs.readFileSync(path.resolve(projectRoot, grammarFile), 'utf8');
-    let result = transform(grammar);
-
-    let grammars = result.procesedGrammars.map(({rules, references, name}) => {
-      rules = rules.map(function(rule) {
-        const ref = references[rule.name] || {};
-        return {
-          name: rule.name,
-          diagram: rule.diagram,
-          usedBy: ref.usedBy,
-          references: ref.references
-        };
-      });
+  let deps = [sources.peg.grammar, diagramStyle, appStyle, baseTemplate];
+  if (targetIsUpToDate(sources.peg.railroad, deps)) {
+    return true;
+  }
 
+  let result = transform(readProjectFile(sources.peg.grammar));
+  let grammars = result.procesedGrammars.map(({rules, references, name}) => {
+    rules = rules.map(function(rule) {
+      const ref = references[rule.name] || {};
       return {
-        name,
-        rules
+        name: rule.name,
+        diagram: rule.diagram,
+        usedBy: ref.usedBy,
+        references: ref.references
       };
     });
 
-    let style = fs.readFileSync(path.resolve(projectRoot, path.join('node_modules', 'grammkit', 'app', 'diagram.css')), 'utf-8') + '\n' +
-                fs.readFileSync(path.resolve(projectRoot, path.join('node_modules', 'grammkit', 'app', 'app.css')), 'utf-8');
-    let data = {
-        title: `Railroad diagram for ${grammarFile}`,
-        style: style,
-        grammars: grammars
-    };
-
-    let output = renderTemplate(data, path.resolve(projectRoot, path.join('node_modules', 'grammkit', 'template', 'viewer.html')));
+    return {name, rules};
+  });
 
-    fs.writeFileSync(path.resolve(projectRoot, railroadFile), output);
-  }
+  let data = {
+      title: `Railroad diagram for ${sources.peg.grammar}`,
```

Comment:
normally just a 2 space indent here

Reply:
Fixed (once I work out how to get the new commit in)